run.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+import os
+from datetime import datetime
+
+import confidence
+import numpy as np
+from lir import CalibratedScorer
+from lir.util import to_odds
+from tqdm import tqdm
+
+from lr_face.data_providers import get_data, make_pairs, Images
+from lr_face.evaluators import evaluate
+from lr_face.experiment_settings import ExperimentSettings
+from lr_face.utils import write_output, parser_setup, process_dataframe
+from params import TIMES
+
+
+def run(args):
+    """
+    Run one or more calibration experiments.
+    The ExperimentSettings class generates a dataframe containing the different parameter combinations called in the
+    command line or in params.py.
+    """
+    experiments_setup = ExperimentSettings(args)
+    parameters_used = experiments_setup.input_parameters  # exclude output columns
+
+    experiment_name = datetime.now().strftime("%Y-%m-%d %H %M %S")
+    params_dict = experiments_setup.data_frame[parameters_used].iloc[0].to_dict()
+
+    plots_dir = os.path.join('output', experiment_name)
+    if not os.path.exists(plots_dir):
+        os.makedirs(plots_dir)
+
+    # load data once
+    real_data_provider = get_data(
+        dataset_callable=params_dict['dataset_callable'],
+        fraction_test=params_dict['fraction_test'],
+    )
+
+    n_experiments = experiments_setup.data_frame.shape[0]
+    for row in tqdm(range(0, n_experiments)):
+        params_dict = experiments_setup.data_frame[parameters_used].iloc[row].to_dict()
+
+        if row < n_experiments / TIMES:
+            # for the first round, make plots
+            make_plots_and_save_as = os.path.join(plots_dir,
+                                                  f"{'_'.join([str(v)[:25] for v in params_dict.values()])}")
+            results = experiment(params_dict, data_provider=real_data_provider,
+                                 make_plots_and_save_as=make_plots_and_save_as)
+        else:
+            results = experiment(params_dict, data_provider=real_data_provider)
+
+        for k, v in results.items():
+            experiments_setup.data_frame.loc[row, k] = v
+
+    experiments_setup.data_frame = process_dataframe(experiments_setup.data_frame)
+    write_output(experiments_setup.data_frame, experiment_name)
+
+
+def experiment(params, data_provider: Images =None, make_plots_and_save_as=None):
+    """
+    Function to run a single experiment with pipeline:
+    DataProvider -> fit model on train data -> fit calibrator on calibrator data -> evaluate test set
+
+    """
+    lr_system = CalibratedScorer(params['scorers'], params['calibrators'])
+    # TODO training will require a different data structure
```

Comment:
Ik denk dat we het (her)trainen van de modellen misschien sowieso het beste los kunnen trekken van deze pipeline, en dat in de repo's van de modellen doen.

Reply:
eens

Reply:
hele training data stuk hier eruit slopen? moeten we wel iets bedenken voor bijhouden met welke data een model gehertraind is